src/core/ComboBox/ComboBox.tsx

```diff
@@ -304,8 +316,27 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     memoizedItems,
   ]);
 
+  const statusIcon = () => {
+    if (icon) {
+      return React.cloneElement(icon, { 'aria-hidden': true });
+    }
+    if (status && message) {
+      return React.cloneElement(StatusIconMap[status](), {
+        'aria-hidden': true,
+      });
+    }
+    return undefined;
+  };
```

Comment:
1. Uppercase function name will allow JSX syntax in usage like `<StatusIcon />`
2. `message` can be checked during rendering
3. Status icons already have `aria-hidden`.

```suggestion
  const StatusIcon = () => {
    if (icon) {
      return React.cloneElement(icon, { 'aria-hidden': true });
    }
    if (status) {
      return StatusIconMap[status]();
    }
    return null;
  };
```